src/System.Management.Automation/engine/parser/AstVisitor.cs

```diff
@@ -567,113 +618,169 @@ protected AstVisitAction CheckScriptBlock(Ast ast)
     public abstract class DefaultCustomAstVisitor : ICustomAstVisitor
     {
         /// <summary/>
-        public virtual object VisitErrorStatement(ErrorStatementAst errorStatementAst) { return null; }
+        public virtual object DefaultVisit(Ast ast) => null;
```

Comment:
Why do we need this if we already have it in the interface?

Reply:
Default implementation of an interface method behaves like an explicit implementation, meaning that it's not visible to an instance of `DefaultCustomAstVisitor`, and you have to cast it to `ICustomAstVisitor` in order to call the method.
So, without this explicit implementation, you will have to write code like this:

```
public virtual object VisitErrorStatement(ErrorStatementAst errorStatementAst) => ((ICustomAstVisitor)this).DefaultVisit(errorStatementAst);
```

Reply:
Thanks for clarify!

> you will have to write code like this

I believe this is implied when working with interfaces.

Also I think now (in the PR context) we want to prefer interfaces.

It worries me that we are creating two entities that were supposed to be one.